scalding-core/src/main/scala/com/twitter/scalding/Stats.scala

```diff
@@ -125,9 +125,15 @@ object RuntimeStats extends java.io.Serializable {
   def addFlowProcess(fp: FlowProcess[_]) {
     val uniqueJobIdObj = fp.getProperty(UniqueID.UNIQUE_JOB_ID)
     if (uniqueJobIdObj != null) {
-      uniqueJobIdObj.asInstanceOf[String].split(",").foreach { uniqueId =>
+      // for speed concern, use a while loop instead of foreach here
+      val splitted = StringUtility.fastSplit(uniqueJobIdObj.asInstanceOf[String], ",")
+      val size = splitted.length
+      var i = 0
+      while (i < size) {
+        val uniqueId = splitted(i)
         logger.debug("Adding flow process id: " + uniqueId)
-        flowMappingStore.put(uniqueId, new WeakReference(fp))
+        flowMappingStore.getOrElse(uniqueId, new WeakReference(fp))
```

Comment:
can you make this:

```scala
if(!flowMappingStore.contains(uniqueId)) {
  logger.debug("Adding flow process id: " + uniqueId)
  flowMappingStore += (uniqueId -> new WeakReference(fp))
}
```
The lazy parameter does an allocation that we could skip for this method if we are calling it a lot.